backend/main.py

```diff
@@ -1,11 +1,73 @@
-from flask import Flask, render_template
+from flask import Flask, jsonify, request, abort
+
+from backend.calc import CharacterStats, Comp, Jobs
+from backend.pps.sch import SchPps
 
 app = Flask(__name__)
 
 
-@app.route('/calc_damage')
+@app.route('/calc_damage', methods=["POST"])
 def main():
-    return "over 9000"
+    """Calculates damage based on input. Accepts and returns JSON. JSON format is as follows:
+    'player': Object
+        'job': string
+        'wd': int
+        'mainstat': int
+        'det': int
+        'crit': int
+        'dh': int
+        'speed': int
+        'ten': int
+        'pie': int
+    'comp': Array
+    """
+    data = request.get_json()
+
+    if not data:
+        return "abort", abort(400)
+
+    # Convert job strings to enum
+    job_string_to_enum = {
```

Comment:
pull this out into a separate helper function, we'll want this for other stuff i assume

Reply:
god damn it i looked at an outdated version